Training6/training-6-sagas-login/src/components/ProtectedPage.js

```diff
@@ -0,0 +1,66 @@
+import React, { useEffect, useState } from "react";
+import { connect } from "react-redux";
+import { Redirect } from "react-router-dom";
+import { onGetUser } from "../actions/index";
+
+ProtectedPage.propTypes = {};
+
+function ProtectedPage({ onGetUser, login, user }) {
+  const tokenCurrent = localStorage.getItem("token");
+  const [redirect, setRedirect] = useState(false);
+  const [isBusy, setBusy] = useState(true);
+  const [data, setData] = useState(user.data);
+
+  useEffect(() => {
+    onGetUser(tokenCurrent);
+    setTimeout(() => setBusy(false), 1000);
+    setData(user.data);
+  }, []);
+
+  function handleLogout() {
+    localStorage.removeItem("token");
+    setRedirect(true);
+    renderRedirect();
+  }
+
+  const renderRedirect = () => {
+    if (redirect === true) {
```

Comment:
This is the wrong pattern to redirect the user to a route, this should be put in `if (!localStorage.getItem("token")) {`